lib/grinder_tools.dart

```diff
@@ -310,50 +310,98 @@ class PubGlobal {
   }
 
   /// Returns whether the given Dart application is installed.
-  bool isInstalled(String packageName) {
+  bool isActivated(String packageName) {
     return list().any((AppVersion app) => app.name == packageName);
   }
 }
 
 /// A Dart command-line application, installed via `pub global activate`.
-class PubApplication {
+abstract class PubApp {
   final String appName;
 
-  bool _installed = false;
+  PubApp._(this.appName);
 
   /// Create a new reference to a pub application; [appName] is the same as the
   /// package name.
-  PubApplication(this.appName);
+  factory PubApp.global(String appName) => new _PubGlobalApp(appName);
 
-  bool isInstalled() {
-    if (_installed) return true;
-    _installed = Pub.global.isInstalled(appName);
-    return _installed;
-  }
+  /// Create a new reference to a pub application; [appName] is the same as the
+  /// package name.
+  factory PubApp.local(String appName) => new _PubLocalApp(appName);
```

Comment:
I appreciate the brevity; right now I think I prefer the additional specificity though even if it makes the ctors a bit longer to reference.